### DISCUSSION TITLE
Problem to separate docker-gen and nginx

### DISCUSSION THREAD

**Discussion by pwFoo** [nginx-proxy/nginx-proxy#1945] [Q&A]
Hello,

I tried your nginx-proxy and it works fine! Thanks for this great image!

I try to separate the nginx and the docker-gen as described at your readme file. I use rancher to manage my containers and can't get it up with the example.

```
 $ docker run --volumes-from nginx \
    -v /var/run/docker.sock:/tmp/docker.sock:ro \
    -v $(pwd):/etc/docker-gen/templates \
    -t jwilder/docker-gen -notify-sighup nginx -watch -only-exposed /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
```

I need to define an command or enty point. Problem is the command part, which can't be executet ad CMD / ENTRY

```
-notify-sighup nginx -watch -only-exposed /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
```

How should I create / start my container?

A prepared docker-gen image to use with nginx would also be great? Any plans to add such a image?

**Comment by joaocc** [comment]
Try removing the "-only-exposed" flag.
When you separate the nginx-gen, it no longer appears in the container list.

Due to the way the nginx.tmpl is setup, it will cause de detection of the CurrentContainer to fail.

**Comment by sgabe** [comment]
You also need to tweak the template as in https://github.com/jwilder/nginx-proxy/pull/750 to detect the proxy container.
